### ISSUE TITLE
Yarn Workspace Support

### ISSUE THREAD

**Issue by jonnii** [FortAwesome/ember-fontawesome#51]
When trying to use this from an application in a yarn workspace I get the following error:

```
ENOENT: no such file or directory, lstat 'c:\project-root\packages\app-in-workspace\node_modules\@fortawesome\fontawesome-svg-core'
```

**Comment by mlwilkerson** [comment]
Could you say more about how you installed the add-on and what version?

**Comment by jonnii** [comment]
We are using a set up similar to the one in [this article on ember engines and yarn workspaces](https://medium.com/square-corner-blog/ember-and-yarn-workspaces-fca69dc5d44a) to have a single repository for a bunch of lazy ember engines and addons. 

In the root of our repo we have a `package.json` that looks like:

```
{
  "private": true,
  "workspaces": [
    "packages/*"
  ]
}
```

I installed the font awesome addon but doing `ember install` in the a subdirectory which contains an ember engine:

```
cd packages/my-engine
ember install @fortawesome/ember-fontawesome
```

I then installed the default free pack and tried to build my application and got the above error. 

Let me know if you need more details.

**Comment by mlwilkerson** [comment]
I don't yet know about ember engines and yarn workspaces, so it's very possible that something about that scenario is different in such a way as to cause this symptom.

The `package.json` in `@fortawesome/ember-fontawesome` does have a dependency on `@fortawesome/fontawesome-svg-core`. So when you do `ember install @fortawesome/ember-fontawesome` in a normal `ember-cli` app, `fontawesome-svg-core` ends up being installed under `node_modules`.

I wonder if in your scenario the top-level app `package.json` needs its own dependency on `fontawesome-svg-core`? What happens if you `yarn add --dev @fortawesome/fontawesome-svg-core` at the top-level?

**Comment by jonnii** [comment]
I don't know if it'll help but when using a yarn workspace your `node_modules` is in the root of your workspace, not relative to your application. e.g. you have:

```
\project-root\node_modules
```

Not

```
\packages\app-in-workspace\node_modules
```

When installing packages though you install them at the _level_ of the app and then yarn does the right thing. Looking at the error message something is trying to resolve another package relative to what (i assume) is a symlink to the top level node_modules package but failing to do so. 

I'll try installing the `svg-core` package explicitly, that may help.

**Comment by mlwilkerson** [comment]
Yes, that could be the key. It’s where the broccoli plugin pulls in that package into the virtual build tree. There’s a good chance it will fail to resolve correctly in your scenario. I could take a closer look later. Would help if you could reproduce the scenario minimally in a GitHub repo.

**Comment by jonnii** [comment]
Let me see what I can do.

**Comment by jonnii** [comment]
Here's a reproduction repo: https://github.com/jonnii/font-awesome-workspace

After cloning it if you do:

```
# install the packages by running yarn install at the root of the cloned repository
# this would, if you had more projects in there, flatten the packages into 
# a top level node_modules 
yarn install 

# start the example ember application, you should now immediately see an error
cd packages/example-app
ember s
```

**Comment by ahawkins** [comment]
@mlwilkerson I opened a similar issue against another addon: https://github.com/kaermorchen/ember-cli-bootstrap-4/issues/34. That contains a reference solution from `ember-bootstrap` for correct file path resolution.

**Comment by andreyfel** [comment]
We are also using yarn workspace in our project. This issue is blocking us from migrating to this addon.

**Comment by bhirsch42** [comment]
Look into the `nohoist` option for yarn workspaces. That's how I'm getting the addon working for the time being.

**Comment by andreyfel** [comment]
The thing is that it will disable the main benefit of yarn workspaces: no duplication of shared node modules. Several projects in our workspace use fontawesome, so, we do want @fortawesome packages to be in the top-level node_modules.

**Comment by jrjohnson** [comment]
Please please please help us get this working! I'm not familiar with yarn workspace at all, but I'm available on Discord to chat about our setup or pair with someone to put a PR together. This probably isn't going to get fixed without someone stepping up so if you know someone or can be that someone....

**Comment by jrjohnson** [comment]
This doesn't fix the problem, but https://github.com/square/find-yarn-workspace-root has a way to find the project root so we could do something like 
``
const workspaceRoot = findWorkspaceRoot(this.app.project.root);
this.nodeModulesPath = workspaceRoot ? workspaceRoot : this.app.project.root;
```

This ensures searching for `node_modules` in the correct place, but when loading the app it's unable to find `index.es.js`, not sure why.

**Comment by jrjohnson** [comment]
I think I solved this in #92. Can anyone using workspaces for real (and not in my contrived way) confirm?
cc/ @jonnii @ahawkins @andreyfel @bhirsch42

**Comment by bhirsch42** [comment]
@jrjohnson Checking it out now

**Comment by bhirsch42** [comment]
🎉 @jrjohnson It works! I no longer need to use `nohoist` on your branch. My test case is an ember app that depends on an ember addon in the same workspace, which in turn depends on `ember-fontawesome`. The app and addon are also have parent directories.

All that to say, I think your solution is probably pretty robust. Thanks a bunch!

**Comment by jrjohnson** [comment]
Released as `0.1.10-1` `-1` because NPM hates me 😿
